mlrun/runtimes/generators.py

```diff
@@ -16,57 +16,86 @@
 import pandas as pd
 import sys
 from copy import deepcopy
-from ..model import RunObject
+from ..model import RunObject, HyperParamOptions
 from ..utils import get_in
 
 
 hyper_types = ["list", "grid", "random"]
 default_max_evals = 10
+default_max_errors = 3
+
+
+def is_hyper_job(spec):
+    param_file = spec.param_file or spec.hyper_options.param_file
+    return param_file or spec.hyperparams
 
 
 def get_generator(spec, execution):
-    tuning_strategy = spec.tuning_strategy
-    hyperparams = spec.hyperparams
-    if not spec.param_file and not hyperparams:
+    options = spec.hyper_options
+    strategy = spec.strategy or options.strategy
+    if not is_hyper_job(spec) or strategy == "custom":
         return None
 
-    if tuning_strategy and tuning_strategy not in hyper_types:
-        raise ValueError(f"unsupported hyperparams type ({tuning_strategy})")
+    hyperparams = spec.hyperparams
+    param_file = spec.param_file or options.param_file
+    if strategy and strategy not in hyper_types:
```

Comment:
Suggest to change `hyper_types` to an enum class (and rename to something more appropriate)